### PR TITLE
tsp, fix on missing test/sample (after projection)

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
fix https://github.com/Azure/autorest.java/issues/2737

fix for project is here
https://github.com/Azure/autorest.java/pull/2736/commits/a9e7189b7adc5e65fa27efaebd9501aba0d821e1 (projection in loadExample)
https://github.com/Azure/autorest.java/pull/2736/commits/e496d90b84c44b68160c3c6ef8748a696d8596f4 (always use projectionSource, if available)

### CODE REVIEW

File: azure-dataplane-tests/src/main/java/com/azure/ai/formrecognizer/documentanalysis/dpg/DocumentModelsAsyncClient.java
Comment (weidongxu-microsoft) [inline azure-dataplane-tests/src/main/java/com/azure/ai/formrecognizer/documentanalysis/dpg/DocumentModelsAsyncClient.java:1]: These due to spec update.

File: javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java [hunk 1]
```diff
@@ -60,7 +61,11 @@ public ClientMethodExampleWriter(ClientMethod method, String clientVarName, Prox
                 .map(nodeVisitor::accept)
                 .collect(Collectors.joining(", "));
 
+        // assertion
         this.imports.add("org.junit.jupiter.api.Assertions");
+        imports.add(LongRunningOperationStatus.class.getName());
+        ClassType.HTTP_HEADER_NAME.addImportsTo(imports, false);
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java:67]: Fix for mssing `HttpHeaderName` import.

File: javagen/src/main/java/com/azure/autorest/template/example/ProtocolExampleWriter.java [hunk 2]
```diff
@@ -81,7 +80,7 @@ public ProtocolExampleWriter(ProtocolExample protocolExample) {
         // assertion
         imports.add("org.junit.jupiter.api.Assertions");
         imports.add(LongRunningOperationStatus.class.getName());
-        imports.add(HttpHeaderName.class.getName());
+        ClassType.HTTP_HEADER_NAME.addImportsTo(imports, false);
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/template/example/ProtocolExampleWriter.java:83]: This does not matter. Just update.

File: typespec-extension/src/operation-utils.ts [hunk 3]
```diff
@@ -51,18 +61,45 @@ export function isKnownContentType(contentTypes: string[]): boolean {
...
-    if (versioning && versioning.getVersions()) {
-      const versions = versioning.getVersions();
-      version = versions[versions.length - 1].value;
+    if (sdkContextApiVersion && !["all", "latest"].includes(sdkContextApiVersion)) {
+      version = sdkContextApiVersion;
+    } else {
+      const versioning = getVersion(program, service.namespace);
+      if (versioning && versioning.getVersions()) {
+        const versions = versioning.getVersions();
+        version = versions[versions.length - 1].value;
```
Comment (XiaofeiCao) [inline typespec-extension/src/operation-utils.ts:90]: Use `getDefaultApiVersion`? They are basically the same, just to align with https://github.com/Azure/autorest.java/blob/5a6afa35f3eeec7fecaa27fd4eb73f551788fafe/typespec-extension/src/code-model-builder.ts#L562-L569

Comment (weidongxu-microsoft) [reply typespec-extension/src/operation-utils.ts]: done

File: javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/template/example/ClientMethodExampleWriter.java:67]: It was there (in a code branch). I just moved it to the top.

**Review by XiaofeiCao — APPROVED** [review]

**Pull request merged**